File: sample_programs/lab5demo.cs
// Dr. Mohan
// Sample file for lab 5: main program to test C# classes "Pirate" and
// "Parrot". See files "pirate.cs" and "parrot.cs"
//
// See file "lab5onefile.cs" for a version that has all three
// classes in a single file.
//
// To compile and run, type:
//          mcs lab5demo.cs pirate.cs parrot.cs  -out:csharpdemo
//          ./csharpdemo

using System;
using PirateNS;
using ParrotNS;
namespace Lab5DemoNS {
  class Lab5Demo {

      static void Main(string[] args) {
         string name;
   
         Console.Write("Pirate's name? ");
         name = Console.ReadLine();
         Pirate pirate = new Pirate(name); // test Pirate constructor
   
         Console.Write("Parrot's name? ");
         name = Console.ReadLine();
         Parrot parrot = new Parrot(name); // test Parrot constructor
   
         // Test "name" and "speak" methods in both Pirate and Parrot:
         Console.WriteLine(pirate.getName()+" says "+pirate.speak());
         Console.WriteLine(pirate.getName()+"'s parrot, "+parrot.getName()+
                          ", says "+ parrot.speak());
      }
   }
}

File: sample_programs/lab5demo.cpp
// Dr. Mohan
// Demonstration program for lab 5. This program is used in
// conjunction with the following files:
//       pirate.h
//       pirate.cpp
//       parrot.h
//       parrot.cpp
//
// For a "single file" version of this example, see file "lab5onefile.cpp".
//
// To compile and execute it, type:
//       g++ lab5demo.cpp pirate.cpp parrot.cpp -o lab5demo
//       ./lab5demo

#include <iostream>
#include <string>
#include "pirate.h"
#include "parrot.h"

using namespace std;

int main() {
   string name;

   cout << "Pirate's name? ";
   cin >> name;
   Pirate pirate(name);

   cout << pirate.getName() << "'s parrot's name? ";
   cin >> name;
   Parrot parrot(name);

   cout << pirate.getName() << " says " << pirate.speak() << endl;
   cout << pirate.getName() << "'s parrot, " << parrot.getName() << ", says " 
        << parrot.speak() << endl;
}
